index.js

```diff
@@ -1,36 +1,44 @@
 const path = require('path');
 const fs = require('fs');
-const {JSDOM} = require('jsdom');
-const {Canvas} = require('canvas');
+const { JSDOM } = require('jsdom');
+const { Canvas } = require('canvas');
 
 // This is probably not the best way to handle Lottie image loading in Node.
 // JSDOM's wrapped `Image` object should work, but attempts resulted in "Image given has not completed loading"
-const createContent = `CVImageElement.prototype.createContent = function() {
-  var assetPath = this.globalData.getAssetsPath(this.assetData);
-  var img = this.img;
-  img.src = assetPath;
-  fs.readFile(assetPath, (err, data) => {
-    if (!err) {
-      img.src = data;
-    }
-    this[err ? 'imageFailed' : 'imageLoaded']();
-  });
-}`;
+const createContent = `
+  function createImgData(assetData) {
+    var path = getAssetsPath(assetData, this.assetsPath, this.path);
+    var ob = {
+      img: null,
+      assetData: assetData,
+    };
+
+    require('canvas').loadImage(path).then((image) => {
+      ob.img = image;
+      this._imageLoaded();
+    }).catch((err) => {
+      ob.img = proxyImage;
+      this._imageLoaded();
+    });
+
+    return ob;
```

Comment:
This returns `ob` before the image is loaded and assigned to ob.img. Is this safe?